public/app/plugins/panel/table/TableCellOptionEditor.tsx

```diff
@@ -0,0 +1,80 @@
+import { merge } from 'lodash';
+import React, { ReactNode } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { TableCellOptions } from '@grafana/schema';
+import { Field, HorizontalGroup, Select, TableCellDisplayMode } from '@grafana/ui';
+
+import { BarGaugeCellOptionsEditor } from './cells/BarGaugeCellOptionsEditor';
+import { ColorBackgroundCellOptionsEditor } from './cells/ColorBackgroundCellOptionsEditor';
+
+const cellDisplayModeOptions = [
+  { value: TableCellDisplayMode.Auto, label: 'Auto' },
+  { value: TableCellDisplayMode.ColorText, label: 'Colored text' },
+  { value: TableCellDisplayMode.ColorBackground, label: 'Colored background' },
+  { value: TableCellDisplayMode.Gauge, label: 'Gauge' },
+  { value: TableCellDisplayMode.JSONView, label: 'JSON View' },
+  { value: TableCellDisplayMode.Image, label: 'Image' },
+];
+
+// Maps display modes to editor components
+interface ComponentMap {
+  [key: string]: Function;
+}
+
+/*
+  Map of display modes to editor components
+  Additional cell types can be placed here
+  ---
+  A cell editor is expected to be a functional
+  component (per the above type) that accepts a
+  sub-options object and an onSubOptions change 
+  callback
+*/
+const displayModeComponentMap: ComponentMap = {
+  [TableCellDisplayMode.Gauge]: BarGaugeCellOptionsEditor,
+  [TableCellDisplayMode.ColorBackground]: ColorBackgroundCellOptionsEditor,
+};
+
+interface Props {
+  value: TableCellOptions;
+  onChange: (v: TableCellOptions) => void;
+}
+
+export const TableCellOptionEditor: React.FC<Props> = ({ value, onChange }) => {
```

Comment:
```suggestion
export function TableCellOptionEditor({ value, onChange }: Props) {
```

Don't use FC type anymore, it's considered deprecated / bad practyice